### ISSUE TITLE
Possibly better to do direct visibility calculation

### ISSUE THREAD

**Issue by steven-murray** [BellaNasirudin/py21cmmc_fg#17]
Instead of doing an FFT on a grid, then interpolating onto baselines, it *may* be better to treat each grid cell as a "point source" and do a direct visbility calculation onto the baselines. I already have code to do this in C (it's very short). What we lose by not doing an FFT in terms of performance, we probably gain from not having to do the interpolation. And it's probably more accurate, and seems to be what more mature simulators do (eg. PRISIM). 

What do you think?

**Comment by BellaNasirudin** [comment]
That is true, interpolation in general is very slow. We can possibly implement this, but won't we still need to grid it and interpolate the centres anyway to find the power spectrum?

**Comment by steven-murray** [comment]
This approach seems to be confirmed as the method of PRISIM and HERAsim.

Yes we'll still need to grid baselines. But that is an independent step.

**Comment by steven-murray** [comment]
I have made an initial stab at this (and some other stuff) in the foreground-rewrite branch (https://github.com/BellaNasirudin/py21cmmc_fg/tree/foreground-rewrite). At this point it doesn't seem to give the correct answer, and is slower than doing the FFT with interpolation by quite a bit (so perhaps it's better to stick with the FFT!).

I've "abandoned" the branch for now, because of these issues. However, if we need this functionality in the future, we can grab the code from there. It also implements the following:

- Each foreground is independent in the way it is defined (i.e. it can have a different sky size and frequency structure).
- Foregrounds are defined on healpix maps so that they can be equal-area. This may not be a good idea though.

**Comment by steven-murray** [comment]
I'm going to close this issue, as I think it is currently too nebulous to do anything. In the end, it will probably be better to use a ready-made visibility simulator rather than implement our own. But that's for far into the future.
